web/js/students_of_class.js

```diff
@@ -245,172 +302,224 @@ function constructStudentTableRow(student) {
     return row;
 }
 
-function deleteStudentRequest() {
+async function deleteStudentRequest() {
     modalDeletion.modal('hide');
     const errorElement = document.querySelector("#error");
-    const student = document.getElementById("studentURL").value;
-    deleteStudent(student)
+    const student = document.querySelector("#studentURL").value;
```

Comment:
I think this doesn't access the right input field there is only one in the edit student modal